src/libraries/System.Data.Common/src/System/Data/DataTableReaderListener.cs

```diff
@@ -7,7 +7,7 @@ namespace System.Data
 {
     internal sealed class DataTableReaderListener
     {
-        private DataTable _currentDataTable;
+        private DataTable? _currentDataTable;
```

Comment:
Thanks.

Interesting... DataTableReaderListener seems to indeed check this for null, but DataTableReader.CurrentDataTable (which it's copied from) seems to always be non-nullable... I'll try to dig a bit deeper, though this is all internal stuff.